crawler.c

```diff
@@ -460,23 +460,27 @@ static void *item_crawler_thread(void *arg) {
     pthread_mutex_unlock(&lru_crawler_lock);
     if (settings.verbose > 2)
         fprintf(stderr, "LRU crawler thread stopping\n");
+    settings.lru_crawler = false;
 
     return NULL;
 }
 
 static pthread_t item_crawler_tid;
 
-int stop_item_crawler_thread(void) {
+int stop_item_crawler_thread(bool wait) {
     int ret;
     pthread_mutex_lock(&lru_crawler_lock);
+    if (do_run_lru_crawler_thread == 0) {
```

Comment:
It is actually important to give the LRU crawler an early out as well.. since the while (crawler_count) {} could run for many minutes, it'll potentially hang the stop routine.

... but that needs some deliberate attention, and it should be safe as-is. I'd like to do this in its own PR. Needs to be careful to hand back client conns before stopping, etc. Had a TODO above for this too but have decided to do later.

Reply:
Makes sense. I was coming at it from the "it can be restarted faster" angle, but needs to be careful about correctness first.

Reply:
This might be a quirk of my personal development habits. This particular change isn't too hard. Since this patch is already large and touches a lot of stuff I try to get more aggressive about pulling out changes... I noticed more bugs creep in and it's much harder to bisect faults when I just "fix this over here" one too many times. When users report bugs often all I get is the before/after version number and an undesired behavior of, "it stopped" :)